test/int_pruf_plant_analysis.py

```diff
@@ -1,102 +1,285 @@
         # Check the pre-processing functions
-        self.check_process_revenue_meter_energy(df)
-        self.check_process_loss_estimates(df)
-        self.check_process_reanalysis_data(df)
+        self.check_process_revenue_meter_energy_daily(df)
+        self.check_process_loss_estimates_daily(df)
+        self.check_process_reanalysis_data_daily(df)
+        
+        # ____________________________________________________________________
+        # Test linear regression model, at monthly time resolution
+        self.analysis = plant_analysis.MonteCarloAEP(self.project, 
+                                                      reanal_products=['merra2', 'era5'],
+                                                      time_resolution = 'M',
+                                                      reg_model = 'lin',
+                                                      reg_temperature = 'N', 
+                                                      reg_winddirection = 'N')
+        # Run Monte Carlo AEP analysis, confirm the results are consistent
+        self.analysis.run(num_sim=10)
+        sim_results = self.analysis.results
+        self.check_simulation_results_lin_monthly(sim_results)
 
-        # Check outlier filtering
-        self.check_filter_outliers()
+        # ____________________________________________________________________
+        # Test linear regression model, at daily time resolution
+        self.analysis = plant_analysis.MonteCarloAEP(self.project, 
+                                                      reanal_products=['merra2', 'era5'],
+                                                      time_resolution = 'D',
+                                                      reg_model = 'lin',
+                                                      reg_temperature = 'N', 
+                                                      reg_winddirection = 'N')
+        # Run Monte Carlo AEP analysis, confirm the results are consistent
+        self.analysis.run(num_sim=10)
+        sim_results = self.analysis.results
+        self.check_simulation_results_lin_daily(sim_results)
 
+        # ____________________________________________________________________
+        # Test GAM regression model (can be used at daily time resolution only)
+        self.analysis = plant_analysis.MonteCarloAEP(self.project, 
+                                                      reanal_products=['merra2', 'era5'],
+                                                      time_resolution = 'D',
+                                                      reg_model = 'gam',
+                                                      reg_temperature = 'N', 
+                                                      reg_winddirection = 'Y')
         # Run Monte Carlo AEP analysis, confirm the results are consistent
-        self.analysis.run(num_sim=2000, reanal_subset=['ncep2', 'merra2', 'erai'])
+        self.analysis.run(num_sim=10)
```

Comment:
When running this set of inputs, I'm getting the following warning na lot:
"SettingWithCopyWarning:
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead"

I couldn't immediately find where it was coming from, but something to keep an eye out for, if you can change the data frame assignment.

Reply:
Thanks for giving me the motivation to find this. This was coming from the use of filters.range_flag. I have fixed that.